x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/store/input/BaseSearchableSnapshotIndexInput.java

```diff
@@ -210,7 +210,7 @@ protected InputStream openSlice(int slice) throws IOException {
      */
     private int getPartNumberForPosition(long position) {
         ensureValidPosition(position);
-        final int part = Math.toIntExact(position / fileInfo.partSize().getBytes());
+        final int part = fileInfo.numberOfParts() == 1 ? 0 : Math.toIntExact(position / fileInfo.partSize().getBytes());
```

Comment:
Snapshots from ES 5 repositories were returning `partSize() == null` sometimes as repositories did not have a non-null partsize configured by default (today it's Long.MAX_VALUE instead for those repos). This would lead to a NPE here.